js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import butterchurn from "butterchurn";
+import butterchurnPresets from "butterchurn-presets";
+import GenWindow from "../GenWindow";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+  }
+  componentDidMount() {
+    const analyserNode = this.props.mediaPlayer._analyser;
+    const gainNode = this.props.mediaPlayer._gainNode;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width,
+        height: this._canvasNode.height
+      }
+    );
+    this.visualizer.connectAudio(gainNode);
+    const presets = butterchurnPresets.getPresets();
```

Comment:
For now we could expose these via the right-click context menu. I've forgotten how they are exposed in real Winamp2. I'll have to download Milkdrop into my VM at some point here.

Reply:
In the original when you press 'L' it loads an overlay that lets you navigate through the current preset director.

space/backspace: for next/prev preset

r: for toggling random vs sequential presets

Reply:
Cool. I'll try to find time to get MilkDrop working locally and try it out.